subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/RegularFileSnapshot.java

```diff
@@ -31,7 +31,11 @@
     private final FileMetadata metadata;
 
     public RegularFileSnapshot(String absolutePath, String name, HashCode contentHash, FileMetadata metadata) {
-        super(absolutePath, name);
+        this(absolutePath, name, contentHash, metadata, false);
+    }
+
+    public RegularFileSnapshot(String absolutePath, String name, HashCode contentHash, FileMetadata metadata, boolean isSymlink) {
```

Comment:
Have you considered introducing new types to represent symlinked files and directories instead of adding a field?

Reply:
I considered that, though I don't see much benefit now. We should introduce the type when we model symlinks as part of the VFS.

Note that we then also will need a different type of CompleteDirectorySnapshot, since a directory with a symlink inside can be viewed in two ways as a snapshot: By resolving the symlink or without resolving it. Both kinds would probably also mean two different Merkle hashes for the same directory.

Reply:
Gaaah. Alright, let's open that can of worms later.